### DISCUSSION TITLE
Getting the queue back out of an AnyScheduler?

### DISCUSSION THREAD

**Discussion by JetForMe** [pointfreeco/combine-schedulers#36] [Q&A] [answered]
I was trying to use this for testing, as shown in your examples, but my coded uses the queue both for `.receive(on:)` and for `DispatchQueue.async` calls (when it calls the delegate we also employ). I can, of course, simply keep the queue reference around and only make the `AnyScheduler` in the `.receive(on:)` calls, but if there's a way to recover the DQ, that would be handy.

**Comment by mbrandonw** [comment, marked as answer]
Hey @JetForMe, this is a good question!

Unfortunately it is not safe to expose a dispatch queue inside the `AnyScheduler` because it's not necessarily always powered by a real life dispatch queue. You can have a `AnySchedulerOf<DispatchQueue>` that secretly is a test scheduler or immediate scheduler under the hood, and so there is not real dispatch queue to expose. 

The best we could do is expose a `var base: Any` that exposes the underlying scheduler, but it would have to be `Any` because the type of the underlying scheduler has been completely forgotten about... which is the point of the `AnyScheduler`. But, if you try casting that `Any` to a `DispatchQueue` it will fail in tests because most likely you are using a test/immediate scheduler.

Having said that, you can still accomplish what you want without getting access to the underlying scheduler. The `Scheduler` protocol has a method called [`.schedule`](https://developer.apple.com/documentation/combine/scheduler/schedule(_:)), and `DispatchQueue`'s implementation simply calls `.async { }`. So you can just do:

```swift
scheduler.schedule { 
  // do work
}
```

Best of all, this works for all schedulers, whether you are really using a dispatch queue, test scheduler, immediate scheduler, etc.

**Reply by JetForMe** [reply]
Oh that's great, that's what I need.
